### PR TITLE
Restructured example notebook folder

### PR DESCRIPTION
**Pull request by ben18785**
Addresses #965 

Have done the following:

- Sorted example notebooks into separate folders
- Updated internal links between notebooks so they now work
- Updated ReadMe so it points to new location of notebooks
- Have updated slow notebooks and error notebooks

### CODE REVIEW

**Comment by ben18785** [comment]
@MichaelClerx Would you be able to take a look at the "run-tests.py" file here? With this new structure of example notebook folders, I think it is not picking up all the files. Also, I am not sure how error-books and slow-books work here but, again, with my new folder structure for the notebooks, it seems to be missing these.

**Comment by ben18785** [comment]
@martinjrobins Could you have a look at the bits I raise above? I don't really get how the testing works at the mo, so not sure what I need to change to rectify things.

File: examples/README.md [hunk 1]
```diff
@@ -8,103 +8,101 @@ relevant code.
...
 
 ## Getting started
-- [Optimisation: First example](./optimisation-first-example.ipynb)
-- [Sampling: First example](./sampling-first-example.ipynb)
-- [Writing a model](./writing-a-model.ipynb)
-- [Writing a custom LogPDF](./writing-a-logpdf.ipynb)
-- [Writing a custom LogPrior](./writing-a-prior.ipynb)
+- [Optimisation: First example](./optimisation/first-example.ipynb)
+- [Sampling: First example](./sampling/first-example.ipynb)
+- [Writing a model](./other-functionality/stats/custom-model.ipynb)
```
Comment (MichaelClerx) [inline examples/README.md]: Why are these 3 in "stats" ?

**Comment by MichaelClerx** [comment]
Hey @ben18785 , made some fixes to the test scripts, which seem OK now.

Could we dispense with the `other-functionality` folder? I'm happy to have plotting, stats, and toy on the same level as optimisation and sampling
Bit worried we'll end up sticking everything else in `other-functionality`!

**Comment by MichaelClerx** [comment]
Hey @ben18785 ! Did you have time to look at the comment above?

**Comment by ben18785** [comment]
Sounds ok with me. In that case, can you go through all example notebooks and check that the links all point to the new folder locations?

**Comment by MichaelClerx** [comment]
Done!

**Pull request merged**